### PR TITLE
Rework ram disk and volatile storage dependencies.

### PR DESCRIPTION
**Pull request by shaper**
Moves TS code cycles from 13 to 12.

RamDisk implementation affected is StorageNG. We could pass the actual memory, rather than a memory provider, but the provider seemed better since it allows for flexibility on the actual backing memory post-initialization. We also could have Runtime contain-a RamDiskMemoryProvider rather than is-a, I didn't feel like the difference was material so went with the latter (and it's easy to change now or later).

Volatile storage implementation affected is non-NG. A shortcut was taken (documented in comment) for volatile storage, rather than doing work to pass the cache service all the way down into the deprecated storage layer.

Part of #1878

### CODE REVIEW

File: src/runtime/ram-disk-memory-provider.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * Code distributed by Google as part of this project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import {VolatileMemory} from './storageNG/drivers/volatile.js';
+
+export interface RamDiskMemoryProvider {
```
Comment (piotrswigon) [inline src/runtime/ram-disk-memory-provider.ts]: Is RamDiskMemoryProvider a best name? WDYT about VolatileMemoryProvider? Seems more high level.

File: src/runtime/manifest.ts [hunk 2]
```diff
@@ -1247,8 +1248,10 @@ ${e.message}
     if (Flags.useNewStorageStack) {
       const storageKey = item['storageKey'] || manifest.createLocalDataStorageKey();
       if (storageKey instanceof RamDiskStorageKey) {
-        const memory = Runtime.getRuntime().getRamDiskMemory();
-        memory.deserialize(entities, storageKey.unique);
+        if (!ramDiskMemoryProvider) {
+          throw new ManifestError(item.location, `Storage is ram disk without memory provider.`);
```
[the remark this answers is not in the document]
Comment (csilvestrini) [reply src/runtime/manifest.ts]: Or maybe something like: "Creating RamDisk stores requires having a RamDiskMemoryProvider"

File: src/runtime/ram-disk-memory-provider.ts (hunk 1 above)
Comment (shaper) [reply src/runtime/ram-disk-memory-provider.ts]: Renamed to VolatileMemoryProvider and moved into `storageNG/drivers/volatile.ts`.

File: src/runtime/manifest.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (shaper) [reply src/runtime/manifest.ts]: Went with `Creating ram disk stores requires having a memory provider.`

**Pull request merged**